js/views/modals/purchase/Payment.js

```diff
@@ -115,8 +117,6 @@ export default class extends BaseVw {
       .fail(jqXhr => {
         openSimpleMessage(app.polyglot.t('purchase.errors.paymentFailed'),
           jqXhr.responseJSON && jqXhr.responseJSON.reason || '');
-      })
-      .always(() => {
         if (this.isRemoved()) return;
```

Comment:
Shouldn't the if removed happen before the simple message is opened?

Reply:
I figured people would want to know if their payment attempt failed, even if they've closed the modal.

Reply:
Gotcha.